### ISSUE TITLE
Android TV: Keyboard automatically open when pressing numeric key in specific conditions 

### ISSUE THREAD

**Issue by freeboub** [react-native-tvos/react-native-tvos#283]
## Description

Keyboard automatically open when pressing numeric key (KEYCODE_1, ... ) after removing an InputText from the View on AndroidTV (not tested on ios, but not sure it makes sens as there is no key number on apple TV)

## React Native version:
0.66.3-3

## Steps To Reproduce
Run following sample. This is just a dummy InputText which is removed from main View onSubmit.
Once you end typing text, the view disappear.
At that point, if you press numerical key on remote control, the keyboard is automatically opened.

As a complementary test, if you initialize show state to false and input numerical Keycode, the issue is not reproduced.

Reproduced on multiple Androidtv devices and on emulator.
Keycode can be simulate with following adb command: 'adb shell input keyevent KEYCODE_1'

## Expected Results
The keyboard should not show at that point.

## code example:

Sample project : https://github.com/freeboub/React-Native-tvos-demo/tree/sample_keyboard_issue

code sample code:

`
import React, {useRef} from 'react';
import {View, TextInput, LogBox} from 'react-native';

const TVEventHandlerView = () => {
  const [show, setShow] = React.useState(true);

  const firstRef = useRef();

  return (
    <>
      {show && (
        <>
          <TextInput
            ref={firstRef}
            defaultValue={'hello'}
            showSoftInputOnFocus={true}
            autoFocus={true}
            editable={true}
            onChangeText={() => {
              console.log('onTextChange');
            }}
            onSubmitEditing={() => {
              console.log('onSubmitEditing');
              setShow(false);
            }}
            placeholder={'Search...'}
          />
        </>
      )}
    </>
  );
};

export default function App() {
  return (
    <View style={{flex: 1}}>
      <TVEventHandlerView />
    </View>
  );
}
`
Additionnal interesting informations:
- If I add a TouchableWithoutFeedback after the InputText the issue is not reproduce (I think focus correctly moves to the TouchableWithoutFeedback)
- If I launch google voice assistant the incorrect behavior disappear.

**Comment by freeboub** [comment]
Try to investigate:
- when the issue happen, android layout inspector confirm that the ReactEditText is well removed from the global layout
- This is reproduced with official react-native 0.66.3
- When the issue present (keyboard opens when it should not), the android Activity doesn't receive onKeyDown ( but correctly receive onKeyUp )

**Comment by douglowder** [comment]
@freeboub please try setting the feature flag for onKeyDown events. See https://github.com/react-native-tvos/react-native-tvos/pull/258 .

**Comment by freeboub** [comment]
Hello @douglowder,
Unfortunnatly it doesn't work... (and notice that shouldDispatchEvent is not called due to dispatchKeyEvent(KeyEvent ev) with action KEY_UP not called ...)
I try to put a lot of logs in react-native (built from source), and I cannot point anything for now...
I tried following things:
- Add traces in ReactActivity onKey and implement dispatchKeyEvent -> nothing the KeyDown is never received in both functions.
- Override public boolean onKeyPreIme (int keyCode, KeyEvent event) in ReactEditText -> it is never called, I also try to implement it in ReactRootView, it is also not called. Notice That it is also not called in 'normal' use case.
- I also try to change options to show and hideKeyboard, also a failure.

I am now suspecting something in lower level of react-native...

If you ave any other clue, I would be happy to test / experiment !
Thank you.

**Comment by freeboub** [comment]
Maybe should I clone this issue on offical repo as it seems to be generic issue ?
I think this is very important with TV integration, but very low criticity on mobile devices ...

**Comment by freeboub** [comment]
Here are some addtionnal informations:
- to workaround the issue I tried to implement by own InputText component (I can share it if needed), but the same issue is reproduced with a custom component, so I think we can asume the issue is not directly linked to EditText management in react-native. It's really looks like a low level issue. Maybe a reference leak ...
- I found an 'ugly' workaround which is working. When a TextInput is blured or unmounted, I launch an empty transparent activty. This activity just call finish onResume... Then the activity stack is correctly cleared and keyboard doesn't take the focus instead of our app. I causes a very small freeze in UI, but It is better than this issue.
